src/sugar.jl

```diff
@@ -1160,3 +1160,76 @@ grad
     return nothing
 end
 
+
+"""
+    seeded_autodiff_thunk(
+        rmode::ReverseModeSplit,
+        dresult,
+        f,
+        ReturnActivity,
+        annotated_args...
+    )
+    
+Call [`autodiff_thunk`](@ref), execute the forward pass, increment output tangent with `dresult`, then execute the reverse pass.
+
+Useful for computing pullbacks / VJPs for functions whose output is not a scalar.
+"""
+function seeded_autodiff_thunk(
+    rmode::ReverseModeSplit{ReturnPrimal},
+    dresult,
+    f::FA,
+    ::Type{RA},
+    args::Vararg{Annotation,N},
+) where {ReturnPrimal,FA<:Annotation,RA<:Annotation,N}
+    forward, reverse = autodiff_thunk(rmode, FA, RA, typeof.(args)...)
+    tape, result, shadow_result = forward(f, args...)
+    if RA <: Active
+        dinputs = only(reverse(f, args..., dresult, tape))
```

Comment:
Just wondering why not just add a vjp, that seems similar to this except doesn't need separate ones for batches

Reply:
Indeed I'm not a fan of the name `seeded_autodiff_thunk` either. `vjp` does convey the notion that input and output have to be vectors, so maybe `pullback` is more generic? What kind of signature do you have in mind?

Reply:
I'm personally fine with the name vjp but at minimum go ahead and add the code and we can iterate on names in parallel

Reply:
wait what do you mean by "add the code"? I thought your request of "adding a vjp" was mostly about naming? what is missing here?

Reply:
We shouldn't have two separate functions here for batched vs not

Reply:
I also thought of doing it via dispatch but how do we handle the case where `dresult` itself is supposed to be an `NTuple` (as opposed to a batch)?